plugins/woocommerce/client/blocks/assets/js/blocks/cart-checkout-shared/payment-methods/payment-methods.tsx

```diff
@@ -45,27 +46,34 @@ const PaymentMethods = ( {
 		};
 	} );
 
-	if ( ! paymentMethodsInitialized ) {
-		return <CheckoutPaymentSkeleton />;
-	}
-
 	const hasPaymentMethods =
 		paymentMethodsInitialized &&
 		Object.keys( availablePaymentMethods ).length > 0;
 	const hasExpressPaymentMethods =
 		expressPaymentMethodsInitialized &&
 		Object.keys( availableExpressPaymentMethods ).length > 0;
 
-	if ( ! hasPaymentMethods && ! hasExpressPaymentMethods ) {
+	if (
+		paymentMethodsInitialized &&
```

Comment:
This logic is confusing me maybe you can explain. The `hasPaymentMethods` const above checks `paymentMethodsInitialized`. If we're not showing a skeleton before `paymentMethodsInitialized`, when is it going to be shown?